elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -400,28 +399,55 @@ private JsonObject getTenantIdFilter() {
 		}
 
 		private Sort getSort(SortField sortField, String idFieldName) {
-			String sortFieldName;
-			if ( sortField.getField() == null ) {
-				switch (sortField.getType()) {
-					case DOC:
-						sortFieldName = "_uid";
-						break;
-					case SCORE:
-						sortFieldName = "_score";
-						break;
-					default:
-						throw LOG.cannotUseThisSortTypeWithNullSortFieldName( sortField.getType() );
-				}
+			if ( sortField instanceof DistanceSortField ) {
+				DistanceSortField distanceSortField = (DistanceSortField) sortField;
+				return new DistanceSort( distanceSortField.getField(),
+						distanceSortField.getCenter(),
+						distanceSortField.getReverse() ? Sorting.DESC : Sorting.ASC );
 			}
 			else {
-				if ( sortField.getField().equals( idFieldName ) ) {
-					sortFieldName = "_uid";
+				String sortFieldName;
+				if ( sortField.getField() == null ) {
+					switch (sortField.getType()) {
+						case DOC:
+							sortFieldName = "_uid";
+							break;
+						case SCORE:
+							sortFieldName = "_score";
+							break;
+						default:
+							throw LOG.cannotUseThisSortTypeWithNullSortFieldName( sortField.getType() );
+					}
 				}
 				else {
-					sortFieldName = sortField.getField();
+					if ( sortField.getField().equals( idFieldName ) ) {
+						sortFieldName = "_uid";
+					}
+					else {
+						sortFieldName = sortField.getField();
+					}
 				}
+				return new Sort( sortFieldName, sortField.getReverse() ? Sorting.DESC : Sorting.ASC );
+			}
+		}
+
+		private void addScriptFields(JsonBuilder.Object query) {
+			if ( isPartOfProjectedFields( ProjectionConstants.SPATIAL_DISTANCE ) ) {
+				// we add a script_field to return the distance
+				query.add( "script_fields",
+						JsonBuilder.object().add( SPATIAL_DISTANCE_FIELD, JsonBuilder.object()
+								.add( "params",
+										JsonBuilder.object()
+												.addProperty( "lat", spatialSearchCenter.getLatitude() )
+												.addProperty( "lon", spatialSearchCenter.getLongitude() )
+								)
+								.addProperty( "script", "doc[\u0027" + spatialFieldName + "\u0027].arcDistanceInKm(lat,lon)" )
```

Comment:
Ok, I suppose that's where Groovy comes into play.

Did you look into [sorting by distance](https://www.elastic.co/guide/en/elasticsearch/guide/current/sorting-by-distance.html) on the ES side? This looks exactly like what we need, so we'd avoid that custom script.